File: engine/model/src/main/scala/tools/packet/alliance/UpdateAllianceJobLevel.scala
package tools.packet.alliance

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class UpdateAllianceJobLevel(private var _allianceId: Int, private var _guildId: Int, private var _characterId: Int, private var _level: Int, private var _jobId: Int) extends PacketInput {
  def allianceId: Int = _allianceId

  def guildId: Int = _guildId

  def characterId: Int = _characterId

  def level: Int = _level

  def jobId: Int = _jobId

  override def opcode(): SendOpcode = SendOpcode.ALLIANCE_OPERATION
}

File: engine/engine-model/src/main/java/tools/packet/alliance/UpdateAllianceJobLevel.java
package tools.packet.alliance;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record UpdateAllianceJobLevel(Integer allianceId, Integer guildId, Integer characterId, Integer level,
                                     Integer jobId) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.ALLIANCE_OPERATION;
   }
}
